lib/winston-logzio.js

```diff
@@ -1,57 +1,45 @@
-var util = require('util');
-var _assign = require('lodash.assign');
-var winston = require('winston');
-var stringifySafe = require('json-stringify-safe');
-var logzioNodejs = require('logzio-nodejs');
+const util = require('util');
+const _assign = require('lodash.assign');
+const winston = require('winston');
+const Transport = require('winston-transport');
+const {createLogger, transports, format} = winston;
+const stringifySafe = require('json-stringify-safe');
+const logzioNodejs = require('logzio-nodejs');
 
-exports = module.exports = winston.transports.Logzio = function (options) {
 
-    this.name = options.name || 'LogzioLogger';
+module.exports = class LogzioWinstonTransport extends Transport {
+    constructor (options) {
+        super(options);
+        this.name = options.name || 'LogzioLogger';
 
-    this.level = options.level || 'info';
+        this.level = options.level || 'info';
 
-    this.logzioLogger = logzioNodejs.createLogger(options);
-};
-var Logzio = winston.transports.Logzio;
+        this.logzioLogger = logzioNodejs.createLogger(options);
+    }
 
-//
-// Inherit from `winston.Transport` so you can take advantage
-// of the base functionality and `.handleExceptions()`.
-//
-util.inherits(Logzio, winston.Transport);
 
-Logzio.prototype.safeToString = function(json) {
-    try {
-        return JSON.stringify(json);
-    }
-    catch(ex) {
-        return stringifySafe(json, null, null, function() { });
-    }
-};
+    log (info, callback) {
+        setImmediate(() => {
+            this.emit('logged', info);
+        });
+        let meta = info[Symbol.for('splat')][0];
+        if (meta instanceof Error) {
+            meta = {error: meta.stack || meta.toString()};
+        }
+        const message = {
+            meta,
+            level: info.level,
+            message: info.message
```

Comment:
What if the log message is a json object? you are ignoring all fields but `message`. Please add all of the fields you get in info and if there is `message` key in info add it under the key `message`.